controllers/submariner/submariner_controller.go

```diff
@@ -116,6 +140,69 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
+	// Ensure we have a secret syncer
+	if instance.Spec.BrokerK8sSecret != "" {
+		if _, ok := r.secretSyncers[instance.Spec.BrokerK8sSecret]; !ok {
+			_, gvr, err := util.ToUnstructuredResource(&corev1.Secret{}, r.config.Client.RESTMapper())
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "error calculating the GVR for the Secret type")
+			}
+			// We can't use files here, we don't have a mounted secret
+			brokerConfig, _, err := resource.GetAuthorizedRestConfigFromData(
+				instance.Spec.BrokerK8sApiServer,
+				instance.Spec.BrokerK8sApiServerToken, // TODO Read the secret
+				instance.Spec.BrokerK8sCA,
+				&rest.TLSClientConfig{Insecure: instance.Spec.BrokerK8sInsecure},
+				*gvr,
+				instance.Spec.BrokerK8sRemoteNamespace)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "error building an authorized RestConfig for the broker")
+			}
+
+			brokerClient, err := dynamic.NewForConfig(brokerConfig)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "error building a dynamic client for the broker")
+			}
+
+			secretSyncer, err := syncer.NewResourceSyncer(
+				&syncer.ResourceSyncerConfig{
+					Name:            "Broker secret syncer",
+					ResourceType:    &corev1.Secret{},
+					SourceClient:    brokerClient,
+					SourceNamespace: instance.Spec.BrokerK8sRemoteNamespace,
+					Direction:       syncer.RemoteToLocal,
+					RestMapper:      r.config.Client.RESTMapper(),
+					Scheme:          r.config.Scheme,
+					Federator: federate.NewCreateOrUpdateFederator(
+						r.config.DynClient, r.config.Client.RESTMapper(), request.Namespace, instance.ClusterName),
+					Transform: func(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
+						secret := from.(*corev1.Secret)
+						reqLogger.V(level.TRACE).Info("Transforming secret", "secret", secret)
+						if saName, ok := secret.ObjectMeta.Annotations["kubernetes.io/service-account.name"]; ok &&
+							saName == broker.ClusterSAName(instance.Spec.ClusterID) {
+							transformedSecret := &corev1.Secret{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: instance.Spec.BrokerK8sSecret,
+								},
+								Type: corev1.SecretTypeOpaque,
+								Data: secret.Data,
+							}
+							reqLogger.V(level.TRACE).Info("Transformed secret", "transformedSecret", transformedSecret)
+							return transformedSecret, false
+						}
+						return nil, false
+					},
+				})
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "error building a resource syncer for secrets")
+			}
+			// TODO skitt handle stopping this when we delete the resource
+			secretSyncer.Start(ctx.Done())
+
+			r.secretSyncers[instance.Spec.BrokerK8sSecret] = secretSyncer
+		}
+	}
```

Comment:
Can we move this into a separate function since this function is already large? (actually I'm surprised this didn't trip the gocyclo linter threshold)